src/accelerate/state.py

```diff
@@ -797,6 +691,79 @@ def default_device(self) -> torch.device:
         else:
             return torch.device("cpu")
 
+    def _prepare_backend(
+        self, cpu: bool = False, sagemaker_dp=False, backend: str = None
+    ) -> tuple[str, DistributedType]:
+        "Prepares any imports needed before initializing the distributed backend and sets `self.backend` properly"
+        if sagemaker_dp:
+            import smdistributed.dataparallel.torch.torch_smddp  # noqa
+
+            backend = "smddp"
+            distributed_type = DistributedType.MULTI_GPU
+        elif int(os.environ.get("LOCAL_RANK", -1)) != -1:
+            if not cpu:
+                if is_mlu_available():
+                    backend = "cncl"
+                    distributed_type = DistributedType.MULTI_MLU
+                elif torch.cuda.is_available():
+                    if backend is None:
+                        backend = "nccl"
+                    distributed_type = DistributedType.MULTI_GPU
+                elif is_npu_available():
+                    backend = "hccl"
+                    distributed_type = DistributedType.MULTI_NPU
+        if backend is None and (
+            int(os.environ.get("LOCAL_RANK", -1)) != -1
+            or get_int_from_env(["PMI_SIZE", "OMPI_COMM_WORLD_SIZE", "MV2_COMM_WORLD_SIZE", "WORLD_SIZE"], 1) > 1
+        ):
+            if not cpu and is_xpu_available():
+                distributed_type = DistributedType.MULTI_XPU
+            else:
+                distributed_type = DistributedType.MULTI_CPU
+            if is_ccl_available() and (
+                get_int_from_env(["CCL_WORKER_COUNT"], 0) > 0 or distributed_type == DistributedType.MULTI_XPU
+            ):
+                if get_ccl_version() >= "1.12":
+                    import oneccl_bindings_for_pytorch  # noqa: F401
+                else:
+                    import torch_ccl  # noqa: F401
+
+                backend = "ccl"
+            elif torch.distributed.is_mpi_available():
+                backend = "mpi"
+            else:
+                backend = "gloo"
+        else:
+            distributed_type = DistributedType.NO
```

Comment:
This makes MLU devices `distributed_type = DistributedType.NO`.
distributed_type may be initialized at function head.

Reply:
Thanks, I'll look into seeing what I did wrong here. Thought I maintained the behavior but it seems like I did not

Reply:
Should be fixed now!